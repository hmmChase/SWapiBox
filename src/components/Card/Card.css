article {
  height: 250px;
  width: 200px;
  background: lightblue
}