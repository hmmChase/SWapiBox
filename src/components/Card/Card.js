import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';
import starFav from '../../images/death-star-fav.svg';
import starUnFav from '../../images/death-star-unfav.svg';

const Card = (props) => {
  const dataKeys = Object.keys(props.cardObj);

  const favoriteIcon = props.favoriteBool
    ? starFav
    : starUnFav;

  const dataElements = dataKeys.map((key, index) => {
    return key === 'name'
      ? <h2 key={index}>{props.cardObj[key]}</h2>
      : <p key={index}>{key}: <span>{props.cardObj[key]}</span></p>;
  });

  return (
    <article>
      <img
        src={favoriteIcon}
        alt="favorite"
        onClick={() => props.updateFavorites(props.cardObj)}
      />
      {dataElements}
    </article>
  );
};

Card.propTypes = {
  favoriteBool: PropTypes.bool,
  cardObj: PropTypes.object,
  updateFavorites: PropTypes.func
};

export default Card;
