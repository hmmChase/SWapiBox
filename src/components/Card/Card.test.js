import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let card;
  beforeEach(() => {
    card = shallow(<Card />);
  });

  it('matches snapshot', () => {
    expect(card).toMatchSnapshot();
  });
});
