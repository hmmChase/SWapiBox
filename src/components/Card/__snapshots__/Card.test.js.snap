// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardObj={
        Object {
            "class": "repulsorcraft",
            "model": "T-16 skyhopper",
            "name": "T-16 skyhopper",
            "passengers": "1",
          }
    }
    favoriteBool={false}
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="favorite"
          onClick={[Function]}
          src="death-star-unfav.svg"
/>,
        Array [
          <h2>
            T-16 skyhopper
</h2>,
          <p>
            model
            : 
            <span>
                        T-16 skyhopper
            </span>
</p>,
          <p>
            class
            : 
            <span>
                        repulsorcraft
            </span>
</p>,
          <p>
            passengers
            : 
            <span>
                        1
            </span>
</p>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "favorite",
          "onClick": [Function],
          "src": "death-star-unfav.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "T-16 skyhopper",
        },
        "ref": null,
        "rendered": "T-16 skyhopper",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "model",
            ": ",
            <span>
              T-16 skyhopper
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "model",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "T-16 skyhopper",
            },
            "ref": null,
            "rendered": "T-16 skyhopper",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "class",
            ": ",
            <span>
              repulsorcraft
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "class",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "repulsorcraft",
            },
            "ref": null,
            "rendered": "repulsorcraft",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "passengers",
            ": ",
            <span>
              1
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "passengers",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="favorite"
            onClick={[Function]}
            src="death-star-unfav.svg"
/>,
          Array [
            <h2>
              T-16 skyhopper
</h2>,
            <p>
              model
              : 
              <span>
                            T-16 skyhopper
              </span>
</p>,
            <p>
              class
              : 
              <span>
                            repulsorcraft
              </span>
</p>,
            <p>
              passengers
              : 
              <span>
                            1
              </span>
</p>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite",
            "onClick": [Function],
            "src": "death-star-unfav.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "T-16 skyhopper",
          },
          "ref": null,
          "rendered": "T-16 skyhopper",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "model",
              ": ",
              <span>
                T-16 skyhopper
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "model",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "T-16 skyhopper",
              },
              "ref": null,
              "rendered": "T-16 skyhopper",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "class",
              ": ",
              <span>
                repulsorcraft
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "class",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "repulsorcraft",
              },
              "ref": null,
              "rendered": "repulsorcraft",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "passengers",
              ": ",
              <span>
                1
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "passengers",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card matches the snapshot when selected as a favorite 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardObj={
        Object {
            "class": "repulsorcraft",
            "model": "T-16 skyhopper",
            "name": "T-16 skyhopper",
            "passengers": "1",
          }
    }
    favoriteBool={false}
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="favorite"
          onClick={[Function]}
          src="death-star-unfav.svg"
/>,
        Array [
          <h2>
            T-16 skyhopper
</h2>,
          <p>
            model
            : 
            <span>
                        T-16 skyhopper
            </span>
</p>,
          <p>
            class
            : 
            <span>
                        repulsorcraft
            </span>
</p>,
          <p>
            passengers
            : 
            <span>
                        1
            </span>
</p>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "favorite",
          "onClick": [Function],
          "src": "death-star-unfav.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "T-16 skyhopper",
        },
        "ref": null,
        "rendered": "T-16 skyhopper",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "model",
            ": ",
            <span>
              T-16 skyhopper
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "model",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "T-16 skyhopper",
            },
            "ref": null,
            "rendered": "T-16 skyhopper",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "class",
            ": ",
            <span>
              repulsorcraft
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "class",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "repulsorcraft",
            },
            "ref": null,
            "rendered": "repulsorcraft",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "passengers",
            ": ",
            <span>
              1
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "passengers",
          ": ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="favorite"
            onClick={[Function]}
            src="death-star-unfav.svg"
/>,
          Array [
            <h2>
              T-16 skyhopper
</h2>,
            <p>
              model
              : 
              <span>
                            T-16 skyhopper
              </span>
</p>,
            <p>
              class
              : 
              <span>
                            repulsorcraft
              </span>
</p>,
            <p>
              passengers
              : 
              <span>
                            1
              </span>
</p>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "favorite",
            "onClick": [Function],
            "src": "death-star-unfav.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "T-16 skyhopper",
          },
          "ref": null,
          "rendered": "T-16 skyhopper",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "model",
              ": ",
              <span>
                T-16 skyhopper
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "model",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "T-16 skyhopper",
              },
              "ref": null,
              "rendered": "T-16 skyhopper",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "class",
              ": ",
              <span>
                repulsorcraft
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "class",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "repulsorcraft",
              },
              "ref": null,
              "rendered": "repulsorcraft",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "passengers",
              ": ",
              <span>
                1
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "passengers",
            ": ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
